pyproject.toml

```diff
@@ -32,7 +32,7 @@ ignore-nested-functions = true
 ignore-semiprivate = true
 ignore-setters = true
 fail-under = 70
-exclude = ["build", "docs", "merlin/io", "tests", "setup.py", "versioneer.py"]
+exclude = ["build", "docs", "merlin/core", "merlin/io", "tests", "setup.py", "versioneer.py"]
```

Comment:
There's not really much reason to add docstrings to the newly defined protocols here, since classes that implement the protocols will have their own docstrings